### PR TITLE
Updates to security overview

### PR DESCRIPTION
**Pull request by it33**

### CODE REVIEW

File: source/overview/security.rst [hunk 1]
```diff
@@ -1,37 +1,91 @@
...
    
+- **Security Features**
+   - Optional features to enhance security on the Mattermost platform. 
+
+To expand on each type: 
+
+Security Updates 
+------
+
+Mattermost software has undergone multiple rounds of security analysis and penetration testing. Security reseachers provide confidential feedback to Mattermost, Inc. using a responsible disclosure policy, allowing for security updates to be provided to the community prior to public disclosure.
```
Comment (lindy65) [inline source/overview/security.rst]: `reseachers' > `researchers`

File: source/overview/security.rst [hunk 2]
```diff
@@ -1,37 +1,91 @@
...
+   - Option to `exclude message contents from push notifications <https://docs.mattermost.com/administration/config-settings.html#push-notification-contents>`_ to comply with strict compliance policies, such as US HIPAA standards.
    
-In addition, Mattermost Enterprise Edition E20 and E10 offer: 
+- Integrity & Audit Controls
+   - By default, Mattermost stores a complete history of messages, including edits and deletes, along with all files uploaded. User interface actions for "deleting" messages, channels and private groups only remove the data from the user interface, the data is retained within your database. 
+   - The `output and archives of server logs <https://docs.mattermost.com/administration/config-settings.html#file-log-directory>`_ can be saved to a directory of your choice. Mattermost server logs plus logs from your web proxy can provide an end-to-end history of system usage.
+   - `Ad hoc compliance reports of messaging by user, date range, and keyword, including edited and deleted messages <https://docs.mattermost.com/administration/compliance.html>`_ are avilable (E20). To protect against unauthorized use, all ad hoc report requests are logged.
```
Comment (lindy65) [inline source/overview/security.rst]: `avilable`  > `available`

File: source/overview/security.rst [hunk 3]
```diff
@@ -1,37 +1,91 @@
...
-- Fast setup of encrypted, private cloud push notifications using `a hosted proxy and pre-compiled mobile applications. <http://docs.mattermost.com/deployment/push.html#hosted-push-notifications-service-hpns>`_
+- Authentication Safeguards 
+   - To protect against brute force attacks, you can set `rate limiting on APIs <https://docs.mattermost.com/administration/config-settings.html#id55>`_, varied by query frequency, memory store size, remote address and headers.    
+   - Session length and session cache can be `configured according to your internal policies <https://docs.mattermost.com/administration/config-settings.html#id33>`_.
+   - Remotely `revoke user sessions <https://docs.mattermost.com/help/settings/account-settings.html#view-and-logout-of-active-sessions>`_ across web, mobile devices and native desktop apps.
+   - Mattermost supports integrated authentcation with `Active Directory and LDAP <https://docs.mattermost.com/deployment/sso-ldap.html>`_ (E10) as well as `Active Directory Federation Services <https://docs.mattermost.com/deployment/sso-saml-adfs.html>`_ and `Okta <https://docs.mattermost.com/deployment/sso-saml-okta.html>`_ via SAML 2.0 (E20)
```
Comment (lindy65) [inline source/overview/security.rst]: `authentcation`  >  `authentication`

File: source/overview/security.rst [hunk 4]
```diff
@@ -1,37 +1,91 @@
...
+   
+Special notes on encryption requirements in highly regulated industries: 
+^^^^^^^ 
 
-- Deploying Mattermost as a HIPPA-compliant communication system requires a deployment team trained on HIPPA-compliance requirements and standards. Some items to note include: 
+- Mobile apps from iTunes or Google Play require special configuration for advanced transmission security
+   - Any mobile app hosted in iTunes or Google Play using push notifications has a moment where the contents of push notifications is decrypted on Apple or Google servers in order to relay the notification to a mobile app. For highly regulated industries that do not allow any moment where confidential data may appear decrypted to a 3rd party, Mattermost offers the ability to prevent the contents from messages from being included in push notifications, by setting `push notification contents <https://docs.mattermost.com/administration/config-settings.html#push-notification-contents>`_ to ``generic``.
```
Comment (lindy65) [inline source/overview/security.rst]: `contents of push notifications is`  > `contents of push notifications are`

File: source/overview/security.rst [hunk 5]
```diff
@@ -1,37 +1,91 @@
...
+ 
+HIPAA compliance 
+^^^^^^^ 
 
-   - If mobile apps from iTunes or Google Play are used with push notifications enabled, it is recommended that `push notification contents <https://docs.mattermost.com/administration/config-settings.html#push-notification-contents>`_ be set to ``generic`` so that confidential data contained in messages does not appear unencrypted to Apple and Google servers relaying notifications to mobile devices.
-   - Transport layer TLS encryption and disk-level encryption on the Mattermost database are options to consider for satisfying HIPPA encryption requirements. 
+- Deploying Mattermost as part of a HIPAA-compliant IT infrastructure requires a deployment team trained on `HIPAA-compliance requirements and standards <http://www.hhs.gov/hipaa/for-professionals/security/laws-regulations/>`_, including administrative safeguards, physical safeguards, organizational requirements and other standards. Techincal safeguards relevant to HIPAA-compliance deployments are described above, please take special note of the deployment notes for "highly regulated industries".
```
Comment (lindy65) [inline source/overview/security.rst]: `Techincal`  > `Technical`

File: source/overview/security.rst [hunk 6]
```diff
@@ -1,37 +1,91 @@
...
+=====================================
+
+Mattermost offers modern communication behind your firewall to keep your data safe, secure, and private.
 
-Mattermost keeps your data safe, secure, and private with enterprise-class data security and control.
+Security Improvements 
+------
 
-Mattermost Enterprise Edition E20, E10, and Team Edition offer: 
+Mattermost is built on an open source platform whose security is under continual review by developers, IT administrators and security researchers accountable for deploying the software in their organizations.
```
Comment (lindy65) [inline source/overview/security.rst]: `Mattermost is built on an open source platform whose security`  > `Mattermost is built on an open source platform, the security of which` ??

**Comment by it33** [comment]
Thanks @lindy65! These all should be fixed now: https://docs.mattermost.com/overview/security.html

**Pull request merged**